src/utils/DependsOnManager.ts

```diff
@@ -30,12 +31,12 @@ export class DependsOnManager {
     this.facet.ref.bind.onRootElement(QueryEvents.newQuery, () => this.handleNewQuery());
 
     if (this.getDependsOn(this.facet.ref)) {
-      $$(this.facet.ref.element).addClass('coveo-hidden');
-      this.setupDependentFacet();
+      this.facet.ref.bind.onRootElement(InitializationEvents.afterComponentsInitialization, () => this.setupDependentFacet());
```

Comment:
Why do we need to delay calling `setupDependentFacet` until the `afterComponentsInitialization`? I'm not clear on how it relates to the PR title: 

> Filter out undefined components when getting instances

Reply:
Oups forgot to change the PR's name, originally I filtered the undefined components triggering the error. 
Changed the solution to something that made more sense.